src/components/alert/alertCenter.svelte

```diff
@@ -109,11 +121,15 @@
         {alert.content}
         <div slot="actions">
           {#each alert.actions as action}
-            <Button
+            <svelte:component
+              this={action.component || Button}
               size={alert.mode === "full" ? "medium" : "small"}
               fab={action.icon && !action.text}
               kind={action.kind || 'filled'}
+              isDisabled={action.isDisabled}
+              isLoading={action.isLoading}
               onClick={() => action.action(alert)}
+              {...action}
```

Comment:
@fallaciousreasoning, I'm getting TS errors here that I've not figured out. Haven't gotten to look too deeply yet, but if you get a moment when you come back online, I'd love your eyes on this.

Reply:
I'd probably just chuck an `as any` at the end - typing it properly probably isn't worth your time here